File: frontend-demo-react/src/app/domain/models/ABC.ts
import { ToastFlavour } from "./notification/flavour";

export abstract class Toastable {

  protected constructor(
    public message:string, 
    public flavour: ToastFlavour) { 

  }

}

export abstract class BaseError extends Toastable { 

  protected constructor(message: string) { 
    super(message, ToastFlavour.Error); 
  }

}

File: backend-demo-dotnet/translate.demo.backend/domain.extensions/Core/Result/ABC.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace domain.extensions.Core.Result
{
  public abstract class BaseError
  {
    public readonly string Message;

    protected BaseError(string error) 
    { 
      Message = error ?? Messages.Core_Error_DefaultError; 
    }

  }

  public class OperationException : Exception
  {

    public OperationException(BaseError error) 
      : base (error.Message) { }

    public static OperationException From(string e)
      => new OperationException(new Error(e));

  }


}
